### DISCUSSION TITLE
How to make UniTask force to block?

### DISCUSSION THREAD

**Discussion by drew-fc** [Cysharp/UniTask#432] [Q&A]
I am doing a network operation to get some data from PlayFab and then my Azure Functions app. The code uses UniTask and looks like this:

```
                var pfAuthResponse = await PlayFabAuthService.Instance.AuthenticateWithDeviceIDAsync(true);
                 await AzureService.Instance.LoginAsync(pfAuthResponse.Result.EntityToken.EntityToken);
```

This works great, however, while testing in development, I want to be able to make these calls block so that I can call them from each scene, without having to load up my splash screen first, so I can more easily test each of my scenes in isolation.

I tried changing the code to this:
```

                var pfAuthResponse = PlayFabAuthService.Instance.AuthenticateWithDeviceIDAsync(true).GetAwaiter().GetResult();
                 AzureService.Instance.LoginAsync(pfAuthResponse.Result.EntityToken.EntityToken).GetAwaiter().GetResult();
```

However, this throws an exception and it looks like it's not possible to GetAwaiter() on a UniTask.
Is there any workaround to force a call to be blocking so I can make my scenes testable in isolation?

**Comment by nicloay** [comment]
Did you find a solution? I also have this problem with converting tasks to synchronous calls.
